samples/TestApplicationShared/App.cs

```diff
@@ -18,35 +18,35 @@ public class App : Application
     {
         public App()
         {
-			// TODO: I believe this has to happen before we select sub systems. Can we
-			// move this safely into Application itself? Is there anything in here
-			// that is platform specific??
-			//
+            // TODO: I believe this has to happen before we select sub systems. Can we
+            // move this safely into Application itself? Is there anything in here
+            // that is platform specific??
+            //
```

Comment:
Well the idea was to make everything overridable - though I'm not sure how useful that would be in practise - but at least in theory I wanted to allow each different part to be replaced if so desired. E.g. suppose someone writes a new layout manager - they can replace the default perspex one by including a .nupkg and registering the service.

Of course this can be done by just re-registering the service after the base `Application` has registered the service, but by making `RegisterServices` overridable it just seemed more composable to me. I'm not stuck on this design though.